src/syscall_mock/mod.rs

```diff
@@ -158,3 +167,47 @@ impl SyscallMock {
         }
     }
 }
+
+#[cfg(test)]
+mod syscall_mock {
+    use super::*;
+    use crate::protocol::Step;
+    use crate::{Context, R};
+    use nix::unistd::Pid;
+    use test_utils::trim_margin;
+
+    #[test]
+    fn resolves_full_path_of_unmocked_commands() -> R<()> {
+        let unmocked_commands = vec![Vec::from("cp")];
+        let syscall_mock = SyscallMock::new(
+            &Context::new()?,
+            Pid::from_raw(123),
+            Protocol::empty(),
+            unmocked_commands.as_slice(),
+        );
+        assert!(syscall_mock.unmocked_commands[0].is_absolute());
+        Ok(())
+    }
+
+    #[test]
+    fn shortens_received_executable_to_file_name_when_reporting_step_error() -> R<()> {
+        let mut syscall_mock = SyscallMock::new(
+            &Context::new()?,
+            Pid::from_raw(123),
+            Protocol::new(vec![Step::from_string("cp")?]),
+            &[],
+        );
+        let mv_command = Command::new("mv")?;
+        syscall_mock.handle_step(mv_command.resolve_executable_path())?;
+        assert_eq!(
+            syscall_mock.result,
+            TestResult::Failure(trim_margin(
+                r#"
+                    |  expected: cp
+                    |  received: mv
+                "#
+            )?)
+        );
+        Ok(())
+    }
```

Comment:
I think I would prefer to test this with an integration test in `tests/path.rs`. I hope that would

1. look a bit nicer and simpler,
2. not rely on implementation details as much.

Reply:
Ok, I replaced these tests with integration tests.